src/test/test-watchdog.c

```diff
@@ -27,12 +27,13 @@ int main(int argc, char *argv[]) {
                 t = 0;
 
         for (i = 0; i < count; i++) {
+                t = watchdog_runtime_wait();
+                log_info("Sleeping " USEC_FMT " microseconds...", t);
+                usleep(t);
```

Comment:
isn't it risky to sleep for the same time (you're at the mercy of the scheduler), any reason to change from t/2?

Reply:
watchdog_runtime_wait() returns half the timeout at most, and takes the last ping into account.

https://github.com/systemd/systemd/blob/v250/src/shared/watchdog.c#L208

Reply:
Also, watchdog_runtime_wait() does not actually wait/sleep itself. It just performs a calculation and returns the result.